float32 p_joint0
float32 p_joint1
float32 p_joint2
float32 p_joint3
float32 p_joint4
float32 p_joint5